Tools/TransformTool/TransformTool.cs

```diff
@@ -363,6 +363,8 @@ public void ProcessInput(ActionMapInput input, Action<InputControl> consumeContr
 #if UNITY_EDITOR
 			if (m_PivotRotation == PivotRotation.Local) // Manipulator does not rotate when in global mode
 				manipulatorTransform.rotation = Quaternion.Slerp(manipulatorTransform.rotation, m_TargetRotation, kLazyFollowRotate * deltaTime);
+
+			UnityEditor.Undo.RecordObjects(Selection.transforms, "EditorVR Manipulator");
```

Comment:
I went back and looked at our prototype code and we had [Undo.IncrementCurrentGroup](https://docs.unity3d.com/ScriptReference/Undo.IncrementCurrentGroup.html). I think this was necessary, so that each move would be handled separately instead of all being grouped together. Can you try out Undos an see if they all end up being grouped together?